pandas/tests/indexing/multiindex/test_iloc.py

```diff
@@ -1,129 +1,97 @@
-        tm.assert_series_equal(rs, xp)
-
-        # corner column
-        rs = mi_int.iloc[2, 2]
-        with catch_warnings(record=True):
-            # First level is int - so use .loc rather than .ix (GH 21593)
-            xp = mi_int.loc[(8, 12), (4, 10)]
-        assert rs == xp
-
-        # this is basically regular indexing
-        rs = mi_labels.iloc[2, 2]
-        with catch_warnings(record=True):
-            xp = mi_labels.ix['j'].ix[:, 'j'].ix[0, 0]
-        assert rs == xp
-
-    def test_frame_getitem_setitem_slice(
-            self, multiindex_dataframe_random_data):
-        frame = multiindex_dataframe_random_data
-        # getitem
-        result = frame.iloc[:4]
-        expected = frame[:4]
-        tm.assert_frame_equal(result, expected)
-
-        # setitem
-        cp = frame.copy()
-        cp.iloc[:4] = 0
-
-        assert (cp.values[:4] == 0).all()
-        assert (cp.values[4:] != 0).all()
-
-    def test_indexing_ambiguity_bug_1678(self):
-        columns = MultiIndex.from_tuples([('Ohio', 'Green'), ('Ohio', 'Red'), (
-            'Colorado', 'Green')])
-        index = MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1), ('b', 2)
-                                        ])
-
-        frame = DataFrame(np.arange(12).reshape((4, 3)), index=index,
-                          columns=columns)
-
-        result = frame.iloc[:, 1]
-        exp = frame.loc[:, ('Ohio', 'Red')]
-        assert isinstance(result, Series)
-        tm.assert_series_equal(result, exp)
-
-    def test_iloc_mi(self):
-        # GH 13797
-        # Test if iloc can handle integer locations in MultiIndexed DataFrame
-
-        data = [['str00', 'str01'], ['str10', 'str11'], ['str20', 'srt21'],
-                ['str30', 'str31'], ['str40', 'str41']]
-
-        mi = MultiIndex.from_tuples(
-            [('CC', 'A'), ('CC', 'B'), ('CC', 'B'), ('BB', 'a'), ('BB', 'b')])
-
-        expected = DataFrame(data)
-        df_mi = DataFrame(data, index=mi)
-
-        result = DataFrame([[df_mi.iloc[r, c] for c in range(2)]
-                            for r in range(5)])
-
-        tm.assert_frame_equal(result, expected)
+@pytest.mark.parametrize('indexer, expected', [
+    (lambda df: df.iloc[0],
+     lambda arr, df: Series(arr[0], index=df.columns, name=(4, 8))),
+    (lambda df: df.iloc[2],
+     lambda arr, df: Series(arr[2], index=df.columns, name=(8, 12))),
+    (lambda df: df.iloc[:, 2],
+     lambda arr, df: Series(arr[:, 2], index=df.index, name=(4, 10))),
+    (lambda df: df.iloc[2, 2],
+     lambda arr, df: arr[2, 2]),
+    (lambda df: df.iloc[[0, 1]],
+     lambda arr, df: df.xs(4, drop_level=False))
+])
+def test_iloc_getitem(indexer, expected):
+    arr = np.random.randn(3, 3)
+    df = DataFrame(arr, columns=[[2, 2, 4], [6, 8, 10]],
+                   index=[[4, 4, 8], [8, 10, 12]])
+
+    result = indexer(df)
+    expected = expected(arr, df)
+
+    try:
+        tm.assert_equal(result, expected)
+    except NotImplementedError:
```

Comment:
what case hits this? would like to more explict here (e.g. not using a try/except)

Reply:
yes let's do that. i am *very* leary anytime have a try/except in a test as they often hide actual errors.